parsec-mock.py

```diff
@@ -1,2 +1,94 @@
 # Copyright 2021 Contributors to the Parsec project.
 # SPDX-License-Identifier: Apache-2.0
+import click
+
+import socket
+import os
+from os import listdir
+from os.path import isfile, join
+
+import base64
+
+from yaml import safe_load
+
+
+@click.command()
+@click.option('--test-folder', default='./testdata', help='Load all tests from folder')
+@click.option('--parsec-socket', default='./parsec.sock', help='Path to parsec unix socket')
+def run_test(test_folder, parsec_socket):
+    print('Mock parsec service listening on unix://{}.'.format(parsec_socket))
+
+    test_cases = load_tests_from_folder(test_folder)
+
+    print('Serving all {} tests in folder {}'.format(len(test_cases),test_folder))
+
+    # Make sure socket doesn't already exist
+    try:
+        os.unlink(parsec_socket)
+    except OSError:
+        if os.path.exists(parsec_socket):
+            print('Error removing old parsec socket, exiting')
+            return
+
+    # Create a unix socket
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind(parsec_socket)
+
+    sock.listen(1)
+
+    while True:
+        connection, client_addr = sock.accept()
+        try:
+            print('Connection received from {}'.format(client_addr))
+            received_data = connection.recv(4096)
+            b64_received_data = base64.b64encode(received_data).decode('ascii')
+            if b64_received_data in test_cases:
+                test_case = test_cases[b64_received_data]
+                print('Received expected request for test case {}'.format(test_case.spec.name))
```

Comment:
*referring to my other comment* Ah I see, this is to be able to access to the name of the test.